src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/RawlsDAO.scala

```diff
@@ -41,6 +41,24 @@ trait RawlsDAO extends LazyLogging with ReportsSubsystemStatus {
 
   def rawlsEntitiesOfTypeUrl(workspaceNamespace: String, workspaceName: String, entityType: String) = FireCloudConfig.Rawls.workspacesUrl + s"/$workspaceNamespace/$workspaceName/entities/$entityType"
 
+  def rawlsQueryEntitiesOfTypeUrl(workspaceNamespace: String, workspaceName: String, entityType: String, query: EntityQuery): String = {
+    val qMap: Map[String, String] = Map(
+      ("page", query.page.toString),
+      ("pageSize", query.pageSize.toString),
+      ("sortField", query.sortField),
+      ("sortDirection", SortDirections.toString(query.sortDirection)))
+    val filteredQMap = query.filterTerms match {
+      case Some(f) => qMap + ("filterTerms" -> f)
+      case _ => qMap
+    }
+    val baseRawlsEntityQueryUrl = FireCloudConfig.Rawls.entityQueryPathFromWorkspace(workspaceNamespace, workspaceName)
+    val baseEntityQueryUri = Uri(baseRawlsEntityQueryUrl)
+    val entityQueryUri = baseEntityQueryUri.
+      withPath(baseEntityQueryUri.path ++ Uri.Path.SingleSlash ++ Uri.Path(entityType)).
+      withQuery(filteredQMap)
+    entityQueryUri.toString()
```

Comment:
I was trying to touch `EntityService` as little as possible due to the weird bugs I saw when the export endpoint was inside of it. I can take a stab at refactoring it to use the dao for some basic uri calculations.